File: sql/plsql/Antlr4cs/PlSqlParserBase.cs
using Antlr4.Runtime;

namespace PlSqlParseTree
{
    public abstract class PlSqlParserBase : Parser
    {
        private bool _isVersion10 = false;
        private bool _isVersion12 = true;

        protected PlSqlParserBase(ITokenStream input)
            : base(input)
        {
        }

        public bool isVersion10() => _isVersion10;

        public bool isVersion12() => _isVersion12;

        public bool setVersion10(bool value) => _isVersion10 = value;

        public bool setVersion12(bool value) => _isVersion12 = value;
    }
}

File: sql/plsql/Go/parser/plsql_parser_base.go
package parser

import (
    "github.com/antlr/antlr4/runtime/Go/antlr"
)

// PlSqlParserBase implementation.
type PlSqlParserBase struct {
    *antlr.BaseParser
    _isVersion12 bool
    _isVersion10 bool
}

//func NewPlSqlParserBase(input TokenStream) *SqlParserBase {
//	b := antlr.NewBaseParser(input)
//	return &PlSqlParserBase{antlr.BaseParser: b, _isVersion12: true, _isVersion10: false}
//}

func (p *PlSqlParserBase) isVersion12() bool {
    return p._isVersion12
}

func (p *PlSqlParserBase) SetVersion12(value bool) {
    p._isVersion12 = value
}

func (p *PlSqlParserBase) isVersion10() bool {
    return p._isVersion10
}

func (p *PlSqlParserBase) SetVersion10(value bool) {
    p._isVersion10 = value
}
